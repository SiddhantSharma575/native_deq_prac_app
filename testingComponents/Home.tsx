import React, { Component } from 'react'
import { Text, TextInput, View } from 'react-native'
import Users from './Users';
import Profile from './Profile';

interface HomeState {
    data:  string
}

export class Home extends Component <{}, HomeState> {
  constructor(props : {}) {
    super(props);
    this.state = {
        data : ''
    }
  }
  
  componentDidMount(): void {
      Users.all().then((data) => {
        console.log("Check Api Data", data)
      })
  }

  change(x : string) {
    // return x * 10;
    this.setState({data : x})
  }
  render() {
    return (
      <View style={{flex : 1, justifyContent:  "center" ,alignItems : "center"}}>
        {/* <TextInput keyboardType="number-pad" testID={"username"} onChangeText={(x) => this.change(Number(x))} style={{backgroundColor : "gray", borderRadius : 15}} placeholder='Enter Number' /> */}
       {/* <TextInput testID="username" onChangeText={(text) => this.change(text)} style={{ borderRadius : 15, width : "100%", height : 44, borderColor : "#000", borderWidth : 0.50}} placeholder="Enter Username" /> */}
        {/* <TextInput testID="password" onChangeText={(text) => this.change(text)} style={{ borderRadius : 15, width : "100%", height : 44, borderColor : "#000", borderWidth : 0.50}} placeholder="Enter Password" /> */}

        <Profile data="Just Testing..." /> 
      </View>
    )
  }
}

export default Home;
