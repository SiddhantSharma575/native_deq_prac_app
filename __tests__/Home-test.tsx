import React from 'react';
import "react-native";
import renderer from "react-test-renderer"
import Home from '../testingComponents/Home';

it("Home Snapshot", () => {
    const snap =renderer.create(<Home />).toJSON()
    expect(snap).toMatchSnapshot()
})

// it("functions and state test care", () => {
//     let HomeData : renderer.ReactTestInstance | null = renderer.create(<Home />).getInstance();
//     // if(HomeData !== null)expect(HomeData.props.change(2)).toEqual(20)
//     if(HomeData !== null) expect(HomeData.state.data).toEqual(0) 
// })