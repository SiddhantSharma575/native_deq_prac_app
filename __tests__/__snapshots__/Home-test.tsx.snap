// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Username"
    style={
      {
        "borderColor": "#000",
        "borderRadius": 15,
        "borderWidth": 0.5,
        "height": 44,
        "width": "100%",
      }
    }
    testID="username"
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Password"
    style={
      {
        "borderColor": "#000",
        "borderRadius": 15,
        "borderWidth": 0.5,
        "height": 44,
        "width": "100%",
      }
    }
    testID="password"
  />
</View>
`;
